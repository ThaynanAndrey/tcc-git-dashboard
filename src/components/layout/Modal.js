import React from 'react';

/**
 * Stateless component with the function to presenting the Modal.
 * 
 * @author Thaynan Nunes
 */
const Modal = ({ children, isFixed }) =>
    (
        <div className={isFixed ? "modal modal-fixed-footer modal-styles" : "modal"}>
            { children }
        </div>
    );

export default Modal;