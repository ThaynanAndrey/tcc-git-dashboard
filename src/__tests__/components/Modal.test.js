import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import Modal from '../../components/layout/Modal';

describe('Test Modal', () => {
  it('renders correctly a modal without props isFixed', () => {
    const tree = shallow(
        <Modal />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders correctly a modal with props isFixed', () => {
    const tree = shallow(
        <Modal isFixed={true}/>
    );

    expect(toJson(tree)).toMatchSnapshot();
  });

  it('the component should have classes modal-fixed-footer and modal-styles when props isFixed is true', () => {
    const tree = shallow(
        <Modal isFixed={true}/>
    );

    expect(tree.find(".modal-fixed-footer.modal-styles").exists()).toBeTruthy();
  });

  it('the component shouldn\'t have classes modal-fixed-footer and modal-styles when props isFixed is false', () => {
    const tree = shallow(
        <Modal isFixed={false}/>
    );

    expect(tree.find(".modal-fixed-footer.modal-styles").exists()).toBeFalsy();
  });
})