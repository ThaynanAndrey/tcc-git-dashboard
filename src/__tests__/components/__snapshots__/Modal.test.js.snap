// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Modal renders correctly a modal with props isFixed 1`] = `
<div
  className="modal modal-fixed-footer modal-styles"
/>
`;

exports[`Test Modal renders correctly a modal without props isFixed 1`] = `
<div
  className="modal"
/>
`;
